AK/Userspace.h

```diff
@@ -20,7 +20,7 @@ class Userspace {
 public:
     Userspace() = default;
 
-    operator bool() const { return m_ptr; }
+    operator bool() const { return static_cast<bool>(m_ptr); }
```

Comment:
Prefer `!= nullptr` as it states intent.

Reply:
That actually doesn't work here because the type of m_ptr is a FlatPtr in one #ifdef path and a pointer in the other.

Reply:
Actually it does work, as long as I move the definitions into the ifdef blocks below. rebase incoming...